### PR TITLE
Issue #31 Adding documentation for developing crc

### LINKED ISSUE #31: Add documentation around how to build CRC for development
We need a documentation around steps around how build CRC locally, so that anyone can compile the code locally i.e.  `developing.adoc`. It should be in the same line as https://docs.okd.io/latest/minishift/contributing/developing.html

### PR DESCRIPTION
**Pull request by LalatenduMohanty**
Fixes #31 

Signed-off-by: Lalatendu Mohanty <lmohanty@redhat.com>

### CODE REVIEW

File: developing.adoc [hunk 1]
```diff
@@ -0,0 +1,45 @@
+
+
+= Developing {project}
```
Comment (robin-owen) [inline developing.adoc]: `{project}` is not defined anywhere.

As I understand it, this project will only have a single name in any case. Considering that, I believe we can simply use a `{crc}` attribute (simply to make referring to the name simpler and less likely to be misrepresented).

File: developing.adoc [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+[[developing-overview]]
+== Overview
+
+The following sections describe how to build and test {project}.
```
Comment (robin-owen) [inline developing.adoc]: `{project}` is not defined.

File: developing.adoc [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+[[develop-prerequisites]]
+== Prerequisites
+
+- Git
+- Make
+- A recent Go distribution (>=1.11)
+
+[NOTE]
+====
+You should be able to develop the project on Linux, Windows or macOS.
```
Comment (robin-owen) [inline developing.adoc]: We should use a serial comma here:

```suggestion
You should be able to develop the project on Linux, Windows, or macOS.
```

File: developing.adoc [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+====
+
+[[set-up-dev-env]]
+== Setting Up the Development Environment
```
Comment (robin-owen) [inline developing.adoc]: I think we should be using sentence case for titles -- it's much simpler to use and understand.

```suggestion
== Setting up the development environment
```

File: developing.adoc [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+[[cloning-repository]]
+=== Cloning the Repository
```
Comment (robin-owen) [inline developing.adoc]:
```suggestion
=== Cloning the repository
```

File: developing.adoc [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+Get the sources from GitHub:
+
+----
+$ git clone https://github.com/code-ready/crc.git
+----
+
+Note: Do not keep the source in your $GOPATH as it will cause `make` to fail. Because CRC uses link:https://github.com/golang/go/wiki/Modules[go modules] and it does not expect the source to be in the $GOPATH.
+
+[[manage-dependencies]]
```
Comment (robin-owen) [inline developing.adoc]: This anchor should accurately reflect the title of the section.

```suggestion
[[dependency-management]]
```

File: developing.adoc [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+== Dependency Management
+
+CRC uses link:https://github.com/golang/go/wiki/Modules[go modules] for dependency management.
+
+Refer below links for more information.
```
Comment (robin-owen) [inline developing.adoc]:
```suggestion
For more information, see the following:
```

File: developing.adoc [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+Get the sources from GitHub:
+
+----
+$ git clone https://github.com/code-ready/crc.git
+----
+
+Note: Do not keep the source in your $GOPATH as it will cause `make` to fail. Because CRC uses link:https://github.com/golang/go/wiki/Modules[go modules] and it does not expect the source to be in the $GOPATH.
+
+[[manage-dependencies]]
+== Dependency Management
```
Comment (robin-owen) [inline developing.adoc]:
```suggestion
== Dependency management
```

File: developing.adoc [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+CRC uses link:https://github.com/golang/go/wiki/Modules[go modules] for dependency management.
```
Comment (robin-owen) [inline developing.adoc]: Capitalize Go:

```suggestion
CRC uses link:https://github.com/golang/go/wiki/Modules[Go modules] for dependency management.
```

File: developing.adoc [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+Refer below links for more information.
+
+. link:https://github.com/golang/go/wiki/Modules[Introduction to go modules]
```
Comment (robin-owen) [inline developing.adoc]: The title for this link is good, as it adds more context. However, we need to capitalize Go:

```suggestion
. link:https://github.com/golang/go/wiki/Modules[Introduction to Go modules]
```

File: developing.adoc [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+. link:https://blog.golang.org/using-go-modules[go modules document about updating dependencies]
```
Comment (robin-owen) [inline developing.adoc]: For this link, I'd use the name of the article:

```suggestion
. link:https://blog.golang.org/using-go-modules[Using Go Modules]
```

File: developing.adoc [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+[[cloning-repository]]
+=== Cloning the Repository
+
+Get the sources from GitHub:
+
+----
+$ git clone https://github.com/code-ready/crc.git
+----
+
+Note: Do not keep the source in your $GOPATH as it will cause `make` to fail. Because CRC uses link:https://github.com/golang/go/wiki/Modules[go modules] and it does not expect the source to be in the $GOPATH.
```
Comment (robin-owen) [inline developing.adoc]: I'd use a block admonition here. Also, do we need to explain the reason behind `make` failing, or would it suffice to simply explain that the source should be outside of your `$GOPATH`?

File: developing.adoc (hunk 1 above)
Comment (LalatenduMohanty) [reply developing.adoc]: @robin-owen Once we have the a release we need to release the docs in a similar way as Minishift. So we need replace this when generating the HTML files. @code-ready/crc-devel  WDYT
I am not sure how to handle the situation now as we will not be releasing this documentation as HTML  till the release. So wondering if we can handle the attribute for GitHub rendering.

File: developing.adoc (hunk 12 above)
Comment (LalatenduMohanty) [reply developing.adoc]: >  would it suffice to simply explain that the source should be outside of your `$GOPATH`?

This enough as `go modules` expect it that way.

File: developing.adoc (hunk 1 above)
Comment (LalatenduMohanty) [reply developing.adoc]: @robin-owen I think we should create an issue for this topic and discuss it there. This discussion should also happen as part of overall documentation discussion.

**Comment by LalatenduMohanty** [comment]
@robin-owen Updated the PR, take a look.

File: README.adoc [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+toc::[]
+
+[[intro-to-crc]]
+== Introduction
+
+This project is focused on bringing a minimal http://github.com/openshift/origin[OpenShift 4.0 or newer] cluster to your local laptop or desktop computer. 
+
+If you are looking for a solution for running OpenShift 3.x, you will need tools such as `oc cluster up`, http://github.com/minishift/minishift[Minishift] or https://developers.redhat.com/products/cdk/overview/[CDK].
+
+Check back for updates, we plan to have something available "soon".
```
Comment (praveenkumar) [inline README.adoc]: `Check back` or `Come back` ? cc @robin-owen

File: README.adoc [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+
+[[documentation]]
+== Documentation
+
+- link:./developing.adoc[Developing CodeReady Containers]
```
Comment (robin-owen) [inline README.adoc]: :+1:

File: README.adoc [hunk 15]
```diff
@@ -0,0 +1,33 @@
+= CodeReady Containers - OpenShift 4 on your Laptop
```
Comment (robin-owen) [inline README.adoc]: :+1: for converting this to AsciiDoc, I was about to open an issue for this myself. :)

File: README.adoc (hunk 13 above)
Comment (robin-owen) [reply README.adoc]: "Check back" is fine -- I'm more concerned about having quotation marks around "soon", personally.

File: developing.adoc [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+[[cloning-repository]]
+=== Cloning the repository
+
+Get the sources from GitHub:
+
+----
+$ git clone https://github.com/code-ready/crc.git
+----
+
+Note: Do not keep the source code in your $GOPATH as link:https://github.com/golang/go/wiki/Modules[Go modules] will cause the `make` to fail.
```
Comment (robin-owen) [inline developing.adoc]: I don't think "Note:" is necessary here unless we want to make this an admonition.

We should mark up `$GOPATH` in monospace.

Minimal rewrite:

```adoc
Do not keep the source code in your `$GOPATH`, as link:https://github.com/golang/go/wiki/Modules[Go modules] will cause `make` to fail.
```

File: developing.adoc [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+Because link:https://github.com/golang/go/wiki/Modules[Go modules]does not expect the source to be in the $GOPATH.
```
Comment (robin-owen) [inline developing.adoc]: I don't believe that this sentence is necessary. This information is available (if needed) in the linked resources regarding Go modules.

File: developing.adoc [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+
+[[dependency-management]]
+== Dependency management
+
+CRC uses link:https://github.com/golang/go/wiki/Modules[Go modules] for dependency management.
```
Comment (robin-owen) [inline developing.adoc]: We should use "CodeReady Containers" rather than the "CRC" acronym in our documentation.

File: README.adoc (hunk 13 above)
Comment (LalatenduMohanty) [reply README.adoc]: @robin-owen Regarding `soon` we can fix it after we have our first release.

**Comment by LalatenduMohanty** [comment]
@praveenkumar @gbraad @anjannath Can you take a look?

**Comment by LalatenduMohanty** [comment]
@gbraad @praveenkumar ping again.

File: README.adoc [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+- link:./developing.adoc[Developing CodeReady Containers]
+
+[[community]]
+== Community
+
+Contributions, questions, and comments are all welcomed and encouraged!
+
+You can reach the community by:
+
+- Joining the #codeready channel on https://freenode.net/[Freenode IRC]
```
Comment (praveenkumar) [inline README.adoc]: :+1:

**Pull request merged**